src/microbe_stage/editor/MicrobeEditorGUI.cs

```diff
@@ -798,15 +798,21 @@ public void UpdateReportTabStatistics(Patch patch)
                 var dataset = speciesPopulationChart.GetDataSet(entry.Key.FormattedName);
 
                 var extinctInPatch = entry.Value <= 0;
+                var extinctEverywhere = snapshot.SpeciesGoneDefinitelyExtinct.Contains(entry.Key);
```

Comment:
As long as external effects are still applied before extinctions are checked it should be fine.

Reply:
This was one of the possibility, but for this to work we need species cloning as underlined below. And, as hh said, I fear this might get a bit expensive. 

There is another potential issue with the fact that we won't be able to (easily) access current state of the species, although this may not truly be a problem for a snapshot (and actually makes sense).

Reply:
Species global population could be split out to a separate class like WorldWideSpeciesInfo or something.

Reply:
It could make sense for more complex stuff, but I'm not sure we'd have much gain from doing this, compared to simply having an extinction list?

Reply:
@Kasterisk probably has a better suggestion on how to store this bit of information for the graphs.

Reply:
You mean, the global population comparison? I'm open to this, but that will requires species cloning to keep the pop count of this time step (not to mark fully extinct pops that only went fully extinct later on).

Reply:
See https://github.com/Revolutionary-Games/Thrive/pull/2427#discussion_r672469159. In general it would be better in the long term, in case there are things in the future that we would like recorded from a species, that data class could come in handy.

Reply:
I guess this could in fact be of use, if we ever want to record things like species structure at this very moment... although I'm still not sure this would be the case. This might still be a bit overkill, but I'll give it a closer look when I'll get back to coding.

Reply:
It seems people want to see a kind of ancestry tree for species. So in the future stuff like generated screenshots could be maybe saved here? I'm sure we can think of some changing species details that could also be stored here.

Reply:
That's precisely what I thought of, but this would currently be accessed from a patch, so it feels... weird. I guess we can have the instance created outside the patch and simply be referenced in all the patch snapshot instances (because it will be discarded anyways after a given count).

But where to store it for the tree? A world snapshot list (we won't go deque as we don't want to discard them)? Obviously, from there it starts getting complicated and outside the scope of the PR, which is why I was cautious at first. But I think I can do what I proposed in the first paragraph, and leave the rest for latter.

Reply:
I mean the tree wouldn't be stored separately. I think that Species should store where they split off from (at least the ID of the parent species) so the tree can be generated quickly when the player wants to look at it.

Reply:
i didn't intend to mean species splitting, but how one species evolve over time (it isn't considered a new species from the game perspective, which could be argued both for and against scientifically). The player's primum thrivum end up being no longer a single-cytoplasm after a while.